### ISSUE TITLE
Simple Custom passengers dataset - regression

### ISSUE THREAD

**Issue by petrhrobar** [timeseriesAI/tsai#483]
HI There,

So I have been playing around with the library for e.g. 5 hours.

Trying to perform simple time-series regression on the ```airpassengers``` dataset.

Could you please point me to what I am doing incorrectly?

```python
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import torch
import random
import seaborn as sns
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.preprocessing import MinMaxScaler

# fastai
from fastai import *
from fastai.text import *
from fastai.text.all import *
from tsai.all import *

flight_data = sns.load_dataset("flights")
flight_data.head(20)

scaler = MinMaxScaler(feature_range=(-1, 1))
# flight_data['passengers'] = scaler.fit_transform(flight_data['passengers'].values.reshape(-1, 1)).flatten()

plt.figure(figsize=(10, 4))
plt.plot(flight_data['passengers'])

def create_inout_sequences(input_data, tw):
    inout_seq = []
    label_seq = []
    L = len(input_data)
    for i in range(L-tw):
        train_seq = input_data[i:i+tw]
        train_label = input_data[i+tw:i+tw+1]
        inout_seq.append(train_seq)
        label_seq.append(train_label)
    return np.array(inout_seq), np.array(label_seq)

data = flight_data['passengers'].values
x, y = create_inout_sequences(data, 15)

src = itemify(x, y)

yy = y.reshape(-1)
xx = x.reshape(-1)



tfms  = [None, [TSRegression()]]
batch_tfms = TSStandardize(by_sample=True, by_var=True)
dls = get_ts_dls(x, yy, tfms=tfms, bs=64)

dls.show_batch()

dls.one_batch()

dls.c
learn = ts_learner(dls, InceptionTime, metrics=[mae, rmse], cbs=ShowGraph())
learn.lr_find()

````

**Comment by oguiza** [comment]
Hi @petrhrobar,
Thanks for sharing this example.
There are at least a couple of issues with your implementation:
1. x needs to have the following shape and dtype: 
   * shape: [n_samples(129) x n_vars(1)  x seq_len(15)]
   * dtype: 'float32' to match model weights
 2. You need to pass splits to get_ts_dls if you want to have a validation set. In this case you should use TimeSplitter.
I've tried this and get a mae around 40 and rmse around 45 without any tuning. 
I believe it's better that you try to do it yourself (you'll learn more). But if you want I can share the code I used.

**Comment by petrhrobar** [comment]
Thanks for that.

I figure to share it as we usually start from a simple csv and not already properly prepared ```X, y``` files.

Could you please also share the full code? I really think a lot of people would benefit and find the ecosystem usefull.

**Comment by oguiza** [comment]
Ok. This is the code I used:
```python
import seaborn as sns
from tsai.all import *

flight_data = sns.load_dataset("flights")
plt.figure(figsize=(10, 4))
plt.plot(flight_data['passengers'])
plt.show()

data = flight_data['passengers'].values
x, y = SlidingWindow(window_len=15)(data)
splits = TimeSplitter()(y)
tfms  = [ToFloat(), ToFloat()] # convert int to float
batch_tfms = TSStandardize()
dls = get_ts_dls(x, y, splits=splits, tfms=tfms, bs=64)
learn = ts_learner(dls, InceptionTime, loss_func=MSELossFlat(), metrics=[mae, rmse], cbs=ShowGraph())
# learn.lr_find()
learn.fit_one_cycle(100, 1e-1)
```
If you want to plot the predictions you can use this: 
```python
preds, target, *_ = learn.get_X_preds(x, y)
valid_preds = preds.clone()
valid_preds[splits[0]] = np.nan
plt.plot(target.flatten().numpy(), color='gray', label='target')
plt.plot(preds.flatten().numpy(), color='orange', label='train preds')
plt.plot(valid_preds.flatten().numpy(), color='purple', label='valid preds')
plt.legend()
plt.show()
```
Here's a link to the [gist](https://colab.research.google.com/gist/oguiza/779aa316000f40bcb67afb6ce6e6e1f6/scratchpad.ipynb) with the results I shared before.

**Comment by petrhrobar** [comment]
Thanks for that.

Just a quick follow-up question. You are not performing a normalization on the data (both for x and y)?

I see ```batch_tfms = TSStandardize()```. However, you are not applying it anywhere.

and when you do: 

```
dls = get_ts_dls(x, y, splits=splits, tfms=tfms, bs=64, batch_tfms = batch_tfms)
dls.one_batch()
```

shows normal values in the level scale?

Am I missing something?

**Comment by oguiza** [comment]
> 

Sorry, that was a miss! You are totally right, and batch_tfms need to be passed to get_ts_dls. I've updated the code and the gist.

**Comment by petrhrobar** [comment]
This still does not seem to solve it. That is my point of asking in the first place.

```python
dls = get_ts_dls(x, y, splits=splits, tfms=tfms, batch_tfms=batch_tfms, bs=64)
dls.one_batch()
```

still return non-normalized data and also during training we may still see that magnitudes of thousand in the loss functions and metrics:

![image](https://user-images.githubusercontent.com/47755107/164211499-82abba2b-76ef-4876-9a98-015af2381cd1.png)

**Comment by oguiza** [comment]
> still return non-normalized data and also during training we may still see that magnitudes of thousand in the loss functions and metrics
The data that needs to be normalized is the X passed to the model.  Whether you normalize it or not it's your choice. The fact that the loss is in the thousands is not an issue. The important outcome is the metric you are interested in.

> That is my point of asking in the first place.
To be honest with you all you said is: Could you please point me to what I am doing incorrectly? It would save us time it next time you clearly state what is the issue you are trying to resolve 🙂
From my standpoint, there's no issue with the code you are using.

**Comment by petrhrobar** [comment]
I was just wondering that it seems like ```TSStandardize()``` in your code is not doing anything.

That is my point.

However, I am very grateful to you for proving an example of loading costume data into the dataloader. 
That solves my issue.

Thanks.

**Comment by oguiza** [comment]
TSStandardize standardizes the x's value not the y's.
To verify that you can do the following: 
```python
xb, yb = dls.one_batch()
print(xb.data.mean(), xb.data.std())
print(yb.data.mean(), yb.data.std())
```
You should see that xb's mean is close to zero and std is close to 1, while that doesn't occur with yb,
